src/main/kotlin/org/rust/lang/core/psi/RsFile.kt

```diff
@@ -56,17 +57,49 @@ class RsFile(
 
     override val containingMod: RsMod get() = getOriginalOrSelf()
 
-    override val crateRoot: RsMod?
+    val cargoProject: CargoProject? get() = cachedData.cargoProject
+    val cargoWorkspace: CargoWorkspace? get() = cachedData.cargoWorkspace
+    val cargoTarget: CargoWorkspace.Target? get() = cachedData.cargoTarget
+    override val crateRoot: RsMod? get() = cachedData.crateRoot
+
+    private val cachedData: CachedData
         get() {
             // [RsModulesIndex.getDeclarationFor] behaves differently depending on whether macros are expanding
-            val key = if (project.macroExpansionManager.isResolvingMacro) CRATE_ROOT_MACROS_KEY else CRATE_ROOT_KEY
+            val key = if (project.macroExpansionManager.isResolvingMacro) CACHED_DATA_MACROS_KEY else CACHED_DATA_KEY
             return CachedValuesManager.getCachedValue(this, key) {
-                CachedValueProvider.Result(doCrateRoot(), rustStructureOrAnyPsiModificationTracker)
+                CachedValueProvider.Result(doGetCachedData(), rustStructureOrAnyPsiModificationTracker)
             }
         }
 
-    private fun doCrateRoot(): RsMod? =
-        superMods.lastOrNull()?.takeIf { it.isCrateRoot }
+    private fun doGetCachedData(): CachedData {
+        val possibleCrateRoot = (superMods.lastOrNull() as? PsiFile)?.originalFile as? RsFile
```

Comment:
Hm, now we extract `originalFile` but previously we didn't do it.
Can it affect something when we are working with a copy of the file (completion?) ?

Reply:
Really I think that this shouldn't affect anything and it's always safe to use an original element of RsFile, but I agree that I shouldn't change this logic in this PR. Anyway, this change isn't meaningful. So I brought back the old behavior